bindings_generator/src/classes.rs

```diff
@@ -112,128 +112,235 @@ pub fn generate_class_constants(class: &GodotClass) -> TokenStream {
+            if v.starts_with(chunk) && is_variant_section {
+                let offset = chunk.len() + 1;
+                let next = &v[offset..];
+                if next.chars().next()?.is_numeric() {
+                    break;
+                }
+                v = next;
+            }
+        }
+        if v != variant {
+            return Some(v);
+        }
     }
 
     None
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn simple_enum() {
+        let actual = generate_enum_variant_name("FooBar", "BAZ");
+        assert_eq!("BAZ", actual);
+    }
+
+    #[test]
+    fn blend_mode() {
+        let actual = generate_enum_variant_name("BlendMode", "BLEND_MODE_INTERPOLATED");
+        assert_eq!("INTERPOLATED", actual);
+    }
+
+    #[test]
+    fn track_type() {
+        let actual = generate_enum_variant_name("TrackType", "TYPE_BEZIER");
+        assert_eq!("BEZIER", actual);
+    }
+
+    #[test]
+    fn viewport_msaa() {
+        // cannot have enum variants starting with a number
+        let actual = generate_enum_variant_name("ViewportMSAA", "VIEWPORT_MSAA_2X");
+        assert_eq!("MSAA_2X", actual);
+    }
+
+    #[test]
+    fn speaker_mode() {
+        // cannot have enum variants starting with a number
+        let actual = generate_enum_variant_name("SpeakerMode", "SPEAKER_MODE_STEREO");
+        assert_eq!("STEREO", actual);
+        let actual = generate_enum_variant_name("SpeakerMode", "SPEAKER_SURROUND_31");
+        assert_eq!("SURROUND_31", actual);
+    }
+
+    #[test]
+    fn attenuation_model() {
+        let actual = generate_enum_variant_name("AttenuationModel", "ATTENUATION_INVERSE_DISTANCE");
+        assert_eq!("INVERSE_DISTANCE", actual);
+    }
+
+    #[test]
+    fn filter_db() {
+        let actual = generate_enum_variant_name("FilterDB", "FILTER_6DB");
+        assert_eq!("FILTER_6DB", actual);
+    }
+
+    #[test]
+    fn saver_flags() {
+        let actual = generate_enum_variant_name("SaverFlags", "FLAG_RELATIVE_PATHS");
+        assert_eq!("RELATIVE_PATHS", actual);
+    }
+
+    #[test]
+    fn poly_end_type() {
+        let actual = generate_enum_variant_name("PolyEndType", "END_POLYGON");
+        assert_eq!("POLYGON", actual);
+    }
+
+    #[test]
+    fn viewport_update_mode() {
+        let actual = generate_enum_variant_name("ViewportUpdateMode", "VIEWPORT_UPDATE_DISABLED");
+        assert_eq!("DISABLED", actual);
+    }
+
+    #[test]
+    fn viewport_clear_mode() {
+        let actual = generate_enum_variant_name("ViewportClearMode", "VIEWPORT_CLEAR_ALWAYS");
+        assert_eq!("ALWAYS", actual);
+    }
+
+    #[test]
+    fn variant_section() {
+        let actual = generate_enum_variant_name("FooBar", "FOOBY_BAZ");
+        assert_eq!("FOOBY_BAZ", actual);
+    }
+
+    /*
+    #[test]
+    fn power_state() {
+        let actual = generate_enum_variant_name("PowerState", "POWERSTATE_ON_BATTERY");
```

Comment:
Maybe this can be handled using a separate check without chunking and comparing the full name in uppercase?